### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by KFilippopolitis**

### CODE REVIEW

File: mipdb/exceptions.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from functools import wraps
 from contextlib import contextmanager
 
-sys.tracebacklimit = 0
```
Comment (KFilippopolitis) [inline mipdb/exceptions.py:6]: This will be reverted

File: mipdb/commands.py [hunk 2]
```diff
@@ -34,83 +32,24 @@
 from mipdb.usecases import UntagDataset
 from mipdb.usecases import ValidateDataset
 
+CONFIG = "/home/config.toml"
```
Comment (ThanKarab) [inline mipdb/commands.py]: `CONFIG_PATH` ?

Also, if you want to change this for dev environment, you could read this path from an env variable, with a default value if it doesn't exist.

File: mipdb/monetdb.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+from typing import Union
+
+import sqlalchemy as sql
+import toml
+
+from mipdb.exceptions import DataBaseError
+
+PRIMARYDATA_TABLE = "primary_data"
+
+CONFIG = "/home/config.toml"
```
Comment (ThanKarab) [inline mipdb/monetdb.py:11]: You can have this in a common `__init__.py` file

File: mipdb/sqlite.py [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+            query = session.query(table)
+            if where_conditions:
+                for col, value in where_conditions.items():
+                    query = query.filter(getattr(table.c, col) == value)
+
+            query.delete(synchronize_session=False)
+            session.commit()
+        except Exception as e:
+            session.rollback()
+            raise
```
Comment (ThanKarab) [inline mipdb/sqlite.py:263]: Is this needed?

Won't this happen by default?

File: mipdb/sqlite.py [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+        finally:
+            session.close()
+        return result
+
+    def drop_table(self, table: str) -> None:
+        session = self.Session()
+        try:
+            table = sql.Table(table, metadata, autoload_with=self._executor)
+            table.drop(bind=self._executor)
+            session.commit()
```
Comment (ThanKarab) [inline mipdb/sqlite.py:247]: Why not `autocommit=true`?

File: mipdb/sqlite_tables.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        )
+        return new_table
+
+    @staticmethod
+    def get_values_from_cdes(cdes):
+        return [{"code": cde.code, "metadata": cde.metadata} for cde in cdes]
+
+    def insert_values(self, values, db):
+        # Needs to be overridden because sqlalchemy and monetdb are not cooperating
+        # well when inserting values to JSON columns
```
Comment (ThanKarab) [inline mipdb/sqlite_tables.py:196]: Forgotten?

File: tests/test_commands.py [hunk 7]
```diff
@@ -330,6 +331,22 @@ def test_invalid_dataset_error_cases(data_model, dataset, exception_message):
     )
 
 
+def test_validate_no_db():
```
Comment (ThanKarab) [inline tests/test_commands.py:334]: `test_validate_no_db` or test no csv folder?

**Review by ThanKarab — APPROVED** [review]

File: tests/test_commands.py (hunk 7 above)
Comment (KFilippopolitis) [reply tests/test_commands.py:334]: This is to test mostly that the Validate folder without database properly raises exception.

File: mipdb/sqlite.py (hunk 4 above)
Comment (KFilippopolitis) [reply mipdb/sqlite.py:263]: No rollback happens by default only when you have autocommit=true

**Pull request merged**